### PR TITLE
hp-Adaptation

### PR DESCRIPTION
**Pull request by MayaTatarelli**
PR to incorporate the work on hp-adaptation

### CODE REVIEW

File: src/flow_solver/flow_solver_cases/naca0012.cpp
Comment (jbrillon) [inline src/flow_solver/flow_solver_cases/naca0012.cpp:69]:
```suggestion
    const bool use_mesh_smoothing = false;
```

File: src/mesh/gmsh_reader.cpp
Comment (jbrillon) [inline src/mesh/gmsh_reader.cpp:712]:
```suggestion
read_gmsh(std::string filename, int requested_grid_order, const bool use_mesh_smoothing)
```

File: src/mesh/gmsh_reader.hpp
Comment (jbrillon) [inline src/mesh/gmsh_reader.hpp:61]:
```suggestion
    read_gmsh(std::string filename, int requested_grid_order = 0, const bool use_mesh_smoothing = true);
```

File: src/parameters/parameters_flow_solver.cpp
Comment (jbrillon) [inline src/parameters/parameters_flow_solver.cpp:203]: Gaussian bump cases need to set this explicitly when they update the `poly_degree` in the loop where the parameters object is being changed then passed to `create_FlowSolver`

File: tests/integration_tests_control_files/euler_integration/naca0012/CMakeLists.txt
Comment (jbrillon) [inline tests/integration_tests_control_files/euler_integration/naca0012/CMakeLists.txt:24]:
```suggestion
        NAME INVISCID_SUBSONIC_NACA0012_LIFT_BASED_HP_ADAPTATION
```

**Review by jbrillon — CHANGES_REQUESTED** [review]
Great work! I've left some comments to address before approval

File: src/parameters/parameters_flow_solver.cpp
Comment (MayaTatarelli) [reply src/parameters/parameters_flow_solver.cpp:203]: Addressed in most recent commit

**Review by jbrillon — APPROVED** [review]
LGTM

File: src/mesh/mesh_adaptation.h
Comment (PranshulThakur) [inline src/mesh/mesh_adaptation.h:54]: It would be better to add a note here about using overintegration if hp-adaptation has issues with convergence.

**Review by PranshulThakur — COMMENTED** [review]
Nice PR to introduce the functionality of choosing between h and p adaptation in PHiLiP. I have added a comment and will approve it soon.

File: src/mesh/mesh_adaptation.cpp
Comment (cpethrick) [inline src/mesh/mesh_adaptation.cpp:56]: Could you add a comment indicating why nothing is done if `h_adaptation` is selected?

**Review by cpethrick — COMMENTED** [review]
Great PR, the new functionality is a good addition to PHiLiP. I have left a few small comments and am happy to approve once they are addressed!

**Review by donovan97 — APPROVED** [review]
Looks good to me!

File: src/mesh/mesh_adaptation.h
Comment (MayaTatarelli) [reply src/mesh/mesh_adaptation.h:54]: Are you sure this is the right place to leave the comment about overintegration?

File: src/mesh/mesh_adaptation.cpp
Comment (jbrillon) [inline src/mesh/mesh_adaptation.cpp:58]: @PranshulThakur Is this still the case?

File: src/parameters/parameters_mesh_adaptation.cpp
Comment (MayaTatarelli) [reply src/parameters/parameters_mesh_adaptation.cpp]: Fixed.

File: src/parameters/parameters_flow_solver.cpp
Comment (MayaTatarelli) [reply src/parameters/parameters_flow_solver.cpp]: FIxed.

File: src/mesh/mesh_adaptation.cpp
Comment (MayaTatarelli) [reply src/mesh/mesh_adaptation.cpp]: Fixed.

**Review by cpethrick — APPROVED** [review]
Looks good, thanks for addressing my comments!

File: src/mesh/mesh_adaptation.h
Comment (PranshulThakur) [reply src/mesh/mesh_adaptation.h:54]: Hmm, ideally we would output the message when mesh_adaptation_type is set to hp_adaptation and the residual does not converge. Maybe you could do that in post processing of the flow solver.

File: src/mesh/mesh_adaptation.cpp
Comment (PranshulThakur) [reply src/mesh/mesh_adaptation.cpp:59]: Nope, this comment can be deleted.

File: src/mesh/mesh_adaptation.cpp
Comment (PranshulThakur) [inline src/mesh/mesh_adaptation.cpp:59]:
```suggestion
```

File: src/mesh/mesh_adaptation.h
Comment (jbrillon) [reply src/mesh/mesh_adaptation.h:54]: Maybe this PR should go through before #164 so that @MayaTatarelli doesn't have a bunch of merge conflicts

Comment (PranshulThakur) [reply src/mesh/mesh_adaptation.h:54]: Agreed, this PR should be merged before #164.

File: src/mesh/mesh_adaptation.cpp
Comment (dougshidong) [reply src/mesh/mesh_adaptation.cpp:59]: Delete, resolve this convo, and merge?

Comment (jbrillon) [inline src/mesh/mesh_adaptation.cpp:59]:
```suggestion
```

**Pull request merged**
